### ISSUE TITLE
Building C library on MacOs Ventura 13.2

### ISSUE THREAD

**Issue by PeterHartfield** [giridhar123/Qt-Rpi4-Mac-Cross-Compile#5]
I have a Raspberry Pi 4 8GB running 32 bit Buster OS (clean install), linux kernel 5.15.84-v7l+, Binutils for Raspbian 2.35.2, GLIBC 2.31 - all steps on the Pi completed successfully.

The MacBook Pro is on Ventura 13.2 (latest release). I have overcome a few problems along the way, but now I'm stuck at the "ct-ng build" stage with the following error "Building C library"...

[INFO ]  =================================================================
[INFO ]  Installing core C gcc compiler
[EXTRA]    Configuring core C gcc compiler
[EXTRA]    Building gcc
[EXTRA]    Installing gcc
[EXTRA]    Housekeeping for core gcc compiler
[EXTRA]       '' --> lib (gcc)   lib (os)
[INFO ]  Installing core C gcc compiler: done in 253.00s (at 17:06)
[INFO ]  =================================================================
[INFO ]  Installing C library
[INFO ]    =================================================================
[INFO ]    Building for multilib 1/1: ''
[EXTRA]      Configuring C library
[EXTRA]      Building C library
[ERROR]      make[3]: *** Cannot open jobserver /var/folders/l7/0g808zpj0_v0y0mb_0p2k82m0000gn/T//GMfifo86008r: No such file or directory.  Stop.
[ERROR]      make[2]: *** [Makefile:470: csu/subdir_lib] Error 2
[ERROR]      make[1]: *** [Makefile:9: all] Error 2
[ERROR]   
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Building for multilib 1/1: '''
[ERROR]  >>        called in step 'Installing C library'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@376]
[ERROR]  >>        called from: glibc_backend_once[scripts/build/libc/glibc.sh@274]
[ERROR]  >>        called from: CT_IterateMultilibs[scripts/functions@1608]
[ERROR]  >>        called from: glibc_main[scripts/build/libc/glibc.sh@40]
[ERROR]  >>        called from: do_libc_main[scripts/build/libc.sh@33]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@697]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      https://crosstool-ng.github.io/docs/known-issues/
[ERROR]  >>
[ERROR]  >> NOTE: You configuration uses non-default patch sets. Please
[ERROR]  >> select 'bundled' as the set of patches applied and attempt
[ERROR]  >> to reproduce this issue. Issues reported with other patch
[ERROR]  >> set selections (none, local, bundled+local) are going to be
[ERROR]  >> closed without explanation.
[ERROR]  >>
[ERROR]  >>  If you feel this is a bug in crosstool-NG, report it at:
[ERROR]  >>      https://github.com/crosstool-ng/crosstool-ng/issues/
[ERROR]  >>
[ERROR]  >>  Make sure your report includes all the information pertinent to this issue.
[ERROR]  >>  Read the bug reporting guidelines here:
[ERROR]  >>      http://crosstool-ng.github.io/support/
[ERROR]   
[ERROR]  (elapsed: 17:16.00)
[17:17] / gmake: *** [/opt/homebrew/bin/ct-ng:261: build] Error 2

I have been through the known issues and upped the file descriptors to 1024, which made no difference. Tried installing glibc and dependencies but keeps saying I need Linux :-( (I'm on MacOS because that's what these instructions are for - right?).

Any suggestions would be most appreciated to get me to the next step, or even better, completion :-)

Thanks in advance,
Peter

**Comment by giridhar123** [comment]
Hello, I did not update this tutorial from some time. I will restore my MacBook and I will follow this tutorial to found new/old misstakes.
Thanks.

**Comment by PeterHartfield** [comment]
Thanks, I would appreciate any help. In the meantime, I have built Qt 6.4.2 and QtCreator 9.0.1 on the Raspberry Pi - no mean feat, took about 12 hrs to build. Now I can at least clone the source from GitHub and build the app on the Pi. 
Cheers,
Peter

**Comment by giridhar123** [comment]
Yes but if the app starts to be complex, compile and test start to be slow too :)

**Comment by giridhar123** [comment]
I have similar problem with glibc 2.35. Try editing the crosstool-ng configuration and set a lower GLIBC version. Then build again.

**Comment by PeterHartfield** [comment]
Thanks for the suggestion, I'll give it a go on the weekend and let you know the results.

**Comment by PeterHartfield** [comment]
I had issues building crosstool-ng on the mac so installed it with brew (1.25.0).
Now having issues building the toolchain as its expecting to get zlib 1.2.12 but I have 1.2.13 installed.
My glibc version is 2.31 - the latest version 2.35 doesn't appear to be supported on M1 MACs (intel 64 bit linux only).
So now I'm not even getting as far as I did last time :-(

**Comment by giridhar123** [comment]
Hello @PeterHartfield I got new about you problem. You can fix that problem in 2 ways. Editing the zlib version manually inside the config file (because the link to download the 1.2.12 version is outdated) and change it to 1.2.13, or, you can build the master branch of the crosstool-ng repository. BTW I recompiled everything (Qt and the cross tools) using GCC 13 and Qt 6.5.1. I think that this time I will build a clone of the SD card of the Raspberry Pi and I will upload all the compile cross tool to avoid this kind of problems. This time it has been hard for me also.
